contractcourt/channel_arbitrator.go

```diff
@@ -932,21 +932,34 @@ func (c *ChannelArbitrator) stateStep(
 		// arbitrating for. If a commitment has confirmed, then we'll
 		// use the set snapshot from the chain, otherwise we'll use our
 		// current set.
-		var htlcs map[HtlcSetKey]htlcSet
-		if confCommitSet != nil {
-			htlcs = confCommitSet.toActiveHTLCSets()
-		} else {
+		var (
+			chainActions ChainActionMap
+			err          error
+		)
+
+		// Normally if we force close the channel locally we will have
+		// no confCommitSet. However when the remote commitment confirms
+		// without us ever broadcasting our local commitment we need to
+		// make sure we cancel all outgoing dust HTLCs as well.
+		if confCommitSet == nil {
```

Comment:
Q: will there be a case when we does local force close but this is not nil?

Reply:
I don't think so, this is only needed in case the remote confirms so we need to look up the remote HTLC set.